src/njoy21/input/GROUPR/Card4/Temp.hpp

```diff
@@ -0,0 +1,41 @@
+struct Temp {
+  using Value_t = std::vector< Quantity< Kelvin > >;
+  static std::string name(){ return "temp"; }
+
+  static std::string description() {
+    return 
+        "The temp argument is a list of temperatures (in Kelvin) to which the\n"
+        "the unresolved resonances are produced.";
+  }
+
+  static bool verify( const Value_t& temps,
+                      const Argument< GROUPR::Card2::Ntemp > & ntemp ){
+    if ( long(temps.size()) != ntemp.value ){
+      Log::info( "Incorrect number of temperatures read in." );
+      return false;
+    }
+    // Make sure all temperatures are positive
+    auto found = std::find_if( temps.begin(), temps.end(),
+                               [](auto& E){ return E < 0.0*kelvin; });
+
+    if ( found != temps.end() ){
+      Log::info( "Negative temperature ({}) found at index {}",
+                    *found, std::distance(temps.begin(), found));
+      return false;
+    } else {
+
+      // Make sure temperatures are in increasing order
+      for( size_t i=1; i < temps.size(); ++i ){
```

Comment:
Good advice, thanks for letting me know. I've implemented it, check me to make sure it is correct.